tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Session.java

```diff
@@ -440,15 +459,63 @@ public Runner runner() {
   private final Graph.Reference graphRef;
 
   private final Object nativeHandleLock = new Object();
-  private long nativeHandle;
+  private TF_Session nativeHandle;
   private int numActiveRuns;
 
+  private static void requireHandle(Pointer handle) {
+    if (handle == null || handle.isNull()) {
+      throw new IllegalStateException("close() has been called on the Session");
+    }
+  }
+
+  private static void resolveHandles(String type, Pointer[] src, PointerPointer dst, int n) {
+    if (src.length != n) {
+      throw new IllegalArgumentException("expected " + n + ", got " + src.length + " " + type);
+    }
+    for (int i = 0; i < n; ++i) {
+      if (src[i] == null || src[i].isNull()) {
+        throw new NullPointerException("invalid " + type + " (#" + i + " of " + n + ")");
```

Comment:
I don't know what was thrown by the original JNI code but I think we should throw ourselves NPE, `IllegalStateException` sounds a better choice. Same thing in `allocate2` below.